hnn_core/optimization.py

```diff
@@ -266,39 +273,60 @@ def _optrun(new_params, opt_params, params, opt_dpls, scale_factor,
     print("Optimization step %d, iteration %d" % (opt_params['cur_step'] + 1,
                                                   opt_params['optiter'] + 1))
 
-    from .parallel_backends import _BACKEND, JoblibBackend
-    if _BACKEND is None:
-        _BACKEND = JoblibBackend(n_jobs=1)
-
     # set parameters
     # tiny negative weights are possible. Clip them to 0.
-    new_params[new_params < 0] = 0
-    for param_name, test_value in zip(opt_params['ranges'].keys(), new_params):
-        params[param_name] = test_value
-
-    # run the simulation, but stop early if possible
-    net = jones_2009_model(params, add_drives_from_params=True)
-    tstop = params['tstop'] = opt_params['opt_end']
-    net._instantiate_drives(n_trials=1, tstop=tstop)
-    avg_dpl = _BACKEND.simulate(net, tstop=tstop, dt=0.025, n_trials=1)[0]
-    avg_dpl = avg_dpl.scale(scale_factor)
+    drive_params_updated = drive_params_updated.copy()
+    drive_params_updated[drive_params_updated < 0] = 0
+    params_dict = dict()
+    for param_name, test_value in zip(opt_params['ranges'].keys(),
+                                      drive_params_updated):
+        params_dict[param_name] = test_value
+
+    # modify drives according to the drive names in the current chunk
+    for drive_idx, drive_name in enumerate(opt_params['inputs']):
+        keys_ampa = fnmatch.filter(list(params_dict.keys()),
+                                   drive_name + '_gbar_ampa_*')
+        keys_nmda = fnmatch.filter(list(params_dict.keys()),
+                                   drive_name + '_gbar_nmda_*')
+        weights_ampa = {key.lstrip(drive_name + '_gbar_ampa_'):
```

Comment:
why do you need `lstrip` ?

Reply:
I guess I could use slice indexing, but that wouldn't be as clear in my opinion. Unless I'm missing something super obvious, the goal here is to isolate the cell type string that can be used as a key for the `weights_ampa` dict.